app/src/main/java/org/oppia/android/app/topic/revision/TopicRevisionFragmentPresenter.kt

```diff
@@ -55,11 +57,22 @@ class TopicRevisionFragmentPresenter @Inject constructor(
       this.viewModel = this@TopicRevisionFragmentPresenter.viewModel
       lifecycleOwner = fragment
     }
+
+    viewModel.subtopicLiveData.observe(fragment) {
+      this.subtopicListSize = it.size
+    }
     return binding.root
   }
 
   override fun onTopicRevisionSummaryClicked(subtopic: Subtopic) {
-    routeToReviewListener.routeToRevisionCard(internalProfileId, topicId, subtopic.subtopicId)
+    routeToReviewListener.routeToRevisionCard(
+      internalProfileId,
+      topicId,
+      subtopic.subtopicId,
+      checkNotNull(subtopicListSize) {
+        throw IllegalArgumentException("Subtopic list size not found.")
+      }
```

Comment:
```suggestion
      checkNotNull(subtopicListSize) { "Subtopic list size not found." }
```

No need for the extra throw.

Reply:
done

Reply:
done